oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -482,17 +481,52 @@ private ImpersonatedCredentials(Builder builder) {
     }
   }
 
+  /**
+   * Gets the universe domain for the credential.
+   *
+   * @return An explicit universe domain if it was explicitly provided, use the universe domain from
+   *     source credentials otherwise
+   */
+  @Override
+  public String getUniverseDomain() throws IOException {
+    if (isExplicitUniverseDomain()) {
```

Comment:
This is for the scenario that customers may create an `ImpersonatedCredentials` object and set `universeDomain` explicitly?

Reply:
Yes.